stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -1144,17 +1144,9 @@ function(add_swift_target_library_single target name)
     target_link_libraries("${target}" INTERFACE ${SWIFTLIB_SINGLE_LINK_LIBRARIES})
   endif()
 
-  # Don't add the icucore target.
-  set(SWIFTLIB_SINGLE_LINK_LIBRARIES_WITHOUT_ICU)
-  foreach(item ${SWIFTLIB_SINGLE_LINK_LIBRARIES})
-    if(NOT "${item}" STREQUAL "icucore")
```

Comment:
I think this was only needed [because of this](https://github.com/apple/swift/pull/40340/files#diff-203086e02ff87aab4512203a96464f0372d2a51c952422cd7eb8962367244eb2L243)?

Reply:
Yes, this was for filtering out icucore which was being added.  Removing the filtering is fine.